File: test_resources/rust/empty_struct_buffers.rs
impl IntoVMBuffers for Empty {
    fn read_from_buffers(_: &mut BytesReader) -> Self {
        Empty
    }

    fn write_to_buffers(&self, _: &mut BytesWriter) {}

    fn skip_in_buffers(_: &mut BytesReader, _: u64) {}
}

File: test_resources/dart/empty_struct_buffers.dart
class EmptyEmplaceToBuffers implements EmplaceToBuffers<Empty> {
  const EmptyEmplaceToBuffers();

  @override
  void read(BytesReader reader, Empty model) {}

  @override
  void write(BytesWriter writer, Empty model) {}

  @override
  void skip(BytesReader reader, int count) {}
}

class EmptyIntoToBuffers implements IntoToBuffers<Empty> {
  const EmptyIntoToBuffers();

  @override
  Empty read(BytesReader reader) {
    return Empty();
  }

  @override
  void write(BytesWriter writer, Empty model) {}

  @override
  void skip(BytesReader reader, int count) {}
}
